spec/app/controllers/course_excluded_exercises_controller/controller_spec.rb

```diff
@@ -0,0 +1,82 @@
+require 'rails_helper'
+
+RSpec.describe CourseExcludedExercisesController, type: :request do
+  let(:request_payload) { 
+    {
+      course_uuid:        given_course_uuid,
+      sequence_number:    given_sequence_number,
+      exclusions:         given_exclusions,
+      request_uuid:       given_request_uuid
+    }
+  }
+
+  let(:given_exclusions) {
+    (given_specific_version_exclusions + given_any_version_exclusions).shuffle
+  }
+
+  let(:given_specific_version_exclusions) {
+    (number_of_exclusions - given_any_version_exclusions.length).times.map{
+      { 'exercise_uuid' => SecureRandom.uuid.to_s }
+    }
+  }
+
+  let(:given_any_version_exclusions) {
+    (Kernel::rand(number_of_exclusions)).times.map{
```

Comment:
Since we are only interested in testing that whatever exclusions exist in the request are passed to the service, I think we can just fix the number of given exclusions and call it good.